python/src/main/python/drivers/piglit-worker.py

```diff
@@ -0,0 +1,418 @@
+
+    # graphicsfuzz_piglit_converter has to skip the first argument since it's meant to be a
+    # standalone script taking arguments from a command line.
+    arglist = list('')
+    arglist.append(str(json_file))
+    if skip_render:
+        arglist.append(NODRAW_ARG)
+
+    shader_test_file = graphicsfuzz_piglit_converter.get_shader_test_from_job(json_file)
+
+    try:
+        log('Creating shader_test file...')
+        graphicsfuzz_piglit_converter.main_helper(arglist)
+    except Exception as ex:
+        log('Could not create shader_test from the given job.')
+        raise ex
+
+    status = STATUS_SUCCESS
+
+    try:
+        run_shader_test(shader_test_file, skip_render)
+    except subprocess.TimeoutExpired:
+        status = STATUS_TIMEOUT
+    except subprocess.CalledProcessError:
+        status = STATUS_CRASH
+
+    # Piglit throws the output PNG render into whatever the current working directory is
+    # (and there's no way to specify a location to write to) - we need to move it to wherever our
+    # output is.
+
+    time.sleep(2)
+
+    if os.path.isfile(os.getcwd() + '/' + PNG_FILENAME):
+        shutil.move(os.getcwd() + '/' + PNG_FILENAME, os.getcwd() + '/' + png_file)
+
+    with gfuzz_common.open_helper(status_file, 'w') as f:
+        f.write(status)
+
+
+def run_shader_test(shader_test_file: str, skip_render: bool):
+    """
+    Runs a shader_test file and logs the output. If the shader runner errors out, the error is
+    logged and then raised to the caller.
+    :param shader_test_file: the shader_test file to run.
+    :param skip_render: whether to skip rendering or not.
+    """
+    shader_runner_cmd = SHADERRUNNER_CMD + [shader_test_file, SHADERRUNNER_ARG_AUTO,
+                                            SHADERRUNNER_ARG_UNIFORMS]
+    if not skip_render:
+        shader_runner_cmd.append(SHADERRUNNER_ARG_PNG)
+    try:
+        log('Exec: ' + shader_test_file + ' with ' + SHADERRUNNER_CMD[0])
+        results = subprocess.run(shader_runner_cmd, timeout=TIMEOUT, check=True)
+    except subprocess.TimeoutExpired as ex:
+        if ex.stdout is not None:
+            log(STDOUT_STR + ex.stdout.decode(encoding='utf-8', errors='ignore'))
+        if ex.stderr is not None:
+            log(STDERR_STR + ex.stderr.decode(encoding='utf-8', errors='ignore'))
+        raise ex
+    except subprocess.CalledProcessError as ex:
+        if ex.stdout is not None:
+            log(STDOUT_STR + ex.stdout.decode(encoding='utf-8', errors='ignore'))
+        if ex.stderr is not None:
+            log(STDERR_STR + ex.stderr.decode(encoding='utf-8', errors='ignore'))
+        log(RETURNCODE_STR + str(ex.returncode))
+        raise ex
+    if results.stdout is not None:
+        log(STDOUT_STR + results.stdout.decode(encoding='utf-8', errors='ignore'))
+    if results.stderr is not None:
+        log(STDERR_STR + results.stderr.decode(encoding='utf-8', errors='ignore'))
+    log(RETURNCODE_STR + str(results.returncode))
```

Comment:
Hopefully most of the above can be removed when you switch to using `runspv.subprocess_helper`